src/io/read_tree_genesis_hdf5.c

```diff
@@ -0,0 +1,1091 @@
+       should be independent)
+     */
+
+    const double maxdiff = 1e-8, maxreldiff = 1e-5; /*numpy.allclose defaults (as of v1.16) */
+#define CHECK_AND_ABORT_UNITS_VS_PARAM_FILE( name, variable, param, absdiff, absreldiff) { \
+        if(AlmostEqualRelativeAndAbs_double(variable, param, absdiff, absreldiff) != EXIT_SUCCESS) { \
+            fprintf(stderr,"Error: Variable %s has value = %g and is different from what is specified in the parameter file = %g\n", \
+                    name, variable, param);                             \
+            return -1;                                                  \
+        }                                                               \
+    }
+
+    CHECK_AND_ABORT_UNITS_VS_PARAM_FILE("Length Unit", lunit, run_params->UnitLength_in_cm, maxdiff, maxreldiff);
+    CHECK_AND_ABORT_UNITS_VS_PARAM_FILE("Velocity Unit", vunit, run_params->UnitVelocity_in_cm_per_s, maxdiff, maxreldiff);
+    CHECK_AND_ABORT_UNITS_VS_PARAM_FILE("Mass Unit", munit, run_params->UnitMass_in_g, maxdiff, maxreldiff);
+
+#undef CHECK_AND_ABORT_UNITS_VS_PARAM_FILE
+
+
+    /* Check that the ID conversion factor is correct */
+    int64_t conv_factor;
+    READ_GENESIS_ATTRIBUTE(gen->h5_fds[start_filenum], "/Header/TreeBuilder", "Temporal_halo_id_value", conv_factor);
+    if(conv_factor != CONVERSION_FACTOR_FOR_GENESIS_UNIQUE_INDEX) {
+        fprintf(stderr,"Error: Expected to find the conversion factor between ID and snapshot + haloindex = %ld "
+                "but instead found = %"PRId64" within the hdf5 file\n",
+                CONVERSION_FACTOR_FOR_GENESIS_UNIQUE_INDEX, conv_factor);
+        return -1;
+    }
+    
+    
+    int num_props_assigned = 0;
+#define ASSIGN_GALAXY_PROPERTY_NAME(propertyname, enumval) {            \
+        snprintf(galaxy_property_names[enumval], MAX_STRING_LEN-1, "%s", propertyname); \
+        num_props_assigned++;                                           \
+    }
+
+    ASSIGN_GALAXY_PROPERTY_NAME("Head", head_enum);
+    ASSIGN_GALAXY_PROPERTY_NAME("Tail", tail_enum);
+    ASSIGN_GALAXY_PROPERTY_NAME("hostHaloID", hosthaloid_enum);
+    ASSIGN_GALAXY_PROPERTY_NAME("Mass_200crit", m200c_enum);
+    ASSIGN_GALAXY_PROPERTY_NAME("Xc", xc_enum);
+    ASSIGN_GALAXY_PROPERTY_NAME("Yc", yc_enum);
+    ASSIGN_GALAXY_PROPERTY_NAME("Zc", zc_enum);
+    ASSIGN_GALAXY_PROPERTY_NAME("VXc", vxc_enum);
+    ASSIGN_GALAXY_PROPERTY_NAME("VYc", vyc_enum);
+    ASSIGN_GALAXY_PROPERTY_NAME("VZc", vzc_enum);
+    ASSIGN_GALAXY_PROPERTY_NAME("Lx", lx_enum);
+    ASSIGN_GALAXY_PROPERTY_NAME("Ly", ly_enum);
+    ASSIGN_GALAXY_PROPERTY_NAME("Lz", lz_enum);
+    ASSIGN_GALAXY_PROPERTY_NAME("sigV", veldisp_enum);
+    ASSIGN_GALAXY_PROPERTY_NAME("ID", mostboundid_enum);
+    ASSIGN_GALAXY_PROPERTY_NAME("npart", len_enum);
+    ASSIGN_GALAXY_PROPERTY_NAME("Vmax", vmax_enum);
+    if(num_props_assigned != num_galaxy_props) {
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Condition 'num_props_assigned!=num_galaxy_props' is always false](https://app.codacy.com/app/jacobseiler/sage-model/pullRequest?prid=4620074)